setuptools/command/easy_install.py

```diff
@@ -2070,19 +2070,36 @@ class ScriptWriter:
     gui apps.
     """
 
-    template = textwrap.dedent(r"""
-        # EASY-INSTALL-ENTRY-SCRIPT: %(spec)r,%(group)r,%(name)r
-        __requires__ = %(spec)r
-        import re
-        import sys
-        from pkg_resources import load_entry_point
-
-        if __name__ == '__main__':
-            sys.argv[0] = re.sub(r'(-script\.pyw?|\.exe)?$', '', sys.argv[0])
-            sys.exit(
-                load_entry_point(%(spec)r, %(group)r, %(name)r)()
-            )
-    """).lstrip()
+    try:
+        from importlib.metadata import distribution  # noqa: F401
+
+        template = textwrap.dedent(r"""
+            # EASY-INSTALL-ENTRY-SCRIPT: %(spec)r,%(group)r,%(name)r
+            __requires__ = %(spec)r
+            import re
+            import sys
+            from importlib.metadata import distribution
+
+            if __name__ == '__main__':
+                sys.argv[0] = re.sub(r'(-script\.pyw?|\.exe)?$', '', sys.argv[0])
+                for entry_point in distribution(%(spec)r).entry_points:
```

Comment:
should this use `entry_points()['console_scripts']` perhaps instead?

Reply:
Wouldn't that enumerate every script for every package?

Reply:
iirc `distribution` already enumerates every package but I might be misremembering

Reply:
I think that's `distributions`:

https://github.com/python/cpython/blob/811e040b6e0241339545c2f055db8259b408802f/Lib/importlib/metadata.py#L498-L512

Reply:
I was surprised this interface isn't nicer :/. Yes, I believe this usage is correct. Probably importlib-metadata should provide some helpers to make this usage simpler, though that won't help for this implementation.